gridt/src/app/settings/change-email/change-email.page.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { of } from 'rxjs';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { FormsModule } from '@angular/forms';
+import { IonicModule } from '@ionic/angular';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { SettingsService } from '../../core/settings.service';
+import { AuthService } from '../../core/auth.service';
+import { Settings } from '../../core/settings.model';
+
+import { ChangeEmailPage } from './change-email.page';
+
+const default_headers = {
+  headers: new HttpHeaders({
+    Authorization: "JWT aksdajskd.asdjknaskdn.asdjknakdnasjd"
+  })
+};
+
+class AuthServiceStub {
+  isLoggedIn$ = of(true);
+  readyAuthentication$ = of(default_headers);
+}
+
+describe('ChangeEmailPage', () => {
+  let component: ChangeEmailPage;
+  let fixture: ComponentFixture<ChangeEmailPage>;
+
+  let setSpy: SettingsService;
+
+  let mock_movement: Settings = {
+    identity: {
+      id: 4,
+      username: "Yo mamma",
+      email: "yomamma@gridt.org",
+      bio: "Don't make me mad",
+      avatar: "abc"
+      }
+    }
+
+  setSpy = jasmine.createSpyObj('SettingsService',
+    {
+      getSettingsFromServer: () => {},
+      the_user_settings$: of(mock_movement),
+      getUserSettings: () => {},
+    }
+  )
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ ChangeEmailPage ],
+      imports: [
+        IonicModule.forRoot(),
+        HttpClientTestingModule,
+        FormsModule
+      ],
+      providers: [
+        { provide: AuthService, useClass: AuthServiceStub }
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ChangeEmailPage);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
```

Comment:
I could imagine that testing that there is a sensible map between the settings that are stored and how they are displayed could help making sure that it works.

Reply:
Am I right in thinking this comment relates more to the settings page instead of the `change-email.page` since this page is only used as a modal.